### PR TITLE
add Infrastructure Lifecycle WG website content

### PR DESCRIPTION
**Pull request by bschaatsbergen**
* Adds home page of the Infrastructure Lifecycle WG
* Copied over our charter
* Contains the start of our glossary

### CODE REVIEW

**Comment by bschaatsbergen** [comment]
> ### ✅ Deploy Preview for _tag-app-delivery_ ready!
> Name	Link
> 🔨 Latest commit	[0548837](https://github.com/cncf/tag-app-delivery/commit/0548837f0a80bf27c4b5da8b2ee71f78337be306)
> 🔍 Latest deploy log	https://app.netlify.com/sites/tag-app-delivery/deploys/66e54abc468e0800086ba86b
> 😎 Deploy Preview	https://deploy-preview-751--tag-app-delivery.netlify.app
> 📱 Preview on mobile	
> Toggle QR Code...
> 
> 
> QR Code
> 
> _Use your smartphone camera to open QR code link._
> _To edit notification comments on pull requests, go to your [Netlify site configuration](https://app.netlify.com/sites/tag-app-delivery/configuration/deploys#deploy-notifications)._

Is it expected that this netlify preview deployment does not render the newly added Infrastructure Lifecycle pages?

**Comment by abangser** [comment]
> Is it expected that the netlify preview deployment does not render the newly added Infrastructure Lifecycle pages?

No, that is definitely showing an issue. But the issue isn't your pages, it is the surfacing of your pages. You can go directly to the URLs like this: https://deploy-preview-751--tag-app-delivery.netlify.app/wgs/infra-lifecycle/

So what you need to add in is the addition of this WG to the different lists etc where the others are.

File: website/content/en/wgs/infra-lifecycle/charter/_index.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
+# Infrastructure Lifecycle Working Group Charter
```
Comment (rynowak) [inline website/content/en/wgs/infra-lifecycle/charter/_index.md:1]: I'm assuming this is a copy-paste from the charter PR?

File: website/content/en/wgs/infra-lifecycle/glossary/_index.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+If you wish to refer to these definitions outside the scope of the working group’s papers, please note that they have been written in the context of both the CNCF and infrastructure management.
+
+## Infrastructure Lifecycle
+The continuous process of designing, deploying, operating, updating, and decommissioning infrastructure resources. This lifecycle covers stages from provisioning to maintenance and eventual retirement, with a focus on resilience, security, sustainability, and automation.
+
+## Infrastructure as Code (IaC)
+A practice where infrastructure is managed and provisioned through machine-readable configuration files, enabling consistency and scalability. Allowing infrastructure to be treated as code, enabling version control, testing, and automation of infrastructure management.
+
+## State
+In infrastructure management, state refers to the current configuration, status, and condition of infrastructure resources at any given point in time. It captures details like resource attributes, dependencies, and operational status, often stored in a state file.
```
Comment (rynowak) [inline website/content/en/wgs/infra-lifecycle/glossary/_index.md:17]: It feels like this needs to be more clear between:

- Desired state (what the user wrote down)
- Actual state (as described below in drift section)
- Stored state (eg: a Terraform state file, which might not be either 😉)

Comment (bschaatsbergen) [reply website/content/en/wgs/infra-lifecycle/glossary/_index.md:17]: I've removed the state definitions, we can work it out later in detail.

**Comment by bschaatsbergen** [comment]
> > Is it expected that the netlify preview deployment does not render the newly added Infrastructure Lifecycle pages?
> 
> No, that is definitely showing an issue. But the issue isn't your pages, it is the surfacing of your pages. You can go directly to the URLs like this: https://deploy-preview-751--tag-app-delivery.netlify.app/wgs/infra-lifecycle/
> 
> So what you need to add in is the addition of this WG to the different lists etc where the others are.

Fixed @abangser! :)

File: website/content/en/wgs/infra-lifecycle/_index.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
+---
+title: Infrastructure Lifecycle Working Group
+list_pages: true
+---
+
+The [charter](./charter/charter.md) describes the goals and activites of this group.
```
Comment (elft3r) [inline website/content/en/wgs/infra-lifecycle/_index.md:6]: In the deploy preview I get a 404 for the charter. It looks like the `charter.md` is not needed.

This is the link that fails: https://deploy-preview-751--tag-app-delivery.netlify.app/wgs/infra-lifecycle/charter/charter.md

```suggestion
The [charter](./charter) describes the goals and activities of this group.
```

Comment (bschaatsbergen) [reply website/content/en/wgs/infra-lifecycle/_index.md]: Fixed, ty!

**Comment by bschaatsbergen** [comment]
@abangser, I've addressed the comments and feedback.

**Review by rynowak — APPROVED** [review]
LGTM

File: README.md [hunk 4]
```diff
@@ -58,3 +58,4 @@ The TAG has created the following working groups to investigate and discuss the
 | [Operator](https://github.com/cncf/tag-app-delivery/tree/main/operator-wg)     |                                                                        | Inactive                                                             |
 | [Artifacts](artifacts-wg/README.md)                                            | [artifacts-wg/README.md#chairs](./artifacts-wg/README.md#chairs)       | [artifacts-wg/README.md#meetings](./artifacts-wg/README.md#meetings) |
 | [App Development](app-development-wg/README.md)                                | [app-development-wg/README.md#chairs](./app-development-wg/README.md#chairs) | [app-development-wg/README.md#meetings](./app-development-wg/README.md#meetings) |
+| [Infrastructure Lifecycle](infra-lifecycle-wg/README.md)                                | [infra-lifecycle-wg/README.md#chairs](./infra-lifecycle-wg/README.md#chairs) | [infra-lifecycle-wg/README.md#meetings](./infra-lifecycle-wg/README.md#meetings) |
```
Comment (abangser) [inline README.md:61]: 💅 Would be a nice to have to keep the formatting for the table in line. So lining up the start of the 2nd and 3rd column in particular.

File: website/content/en/wgs/infra-lifecycle/glossary/_index.md [hunk 5]
```diff
@@ -0,0 +1,14 @@
+---
+title: Glossary
```
Comment (abangser) [inline website/content/en/wgs/infra-lifecycle/glossary/_index.md:2]: I wonder if a goal would be to combine and have a top level glossary?

**Review by abangser — DISMISSED** [review]
I am not a binding acceptance to merge this, but shared a few non-blocking comments to think about!

File: README.md (hunk 4 above)
Comment (bschaatsbergen) [reply README.md]: Fixed

File: website/content/en/wgs/infra-lifecycle/glossary/_index.md (hunk 5 above)
Comment (bschaatsbergen) [reply website/content/en/wgs/infra-lifecycle/glossary/_index.md:2]: That's something I'm happy to address in the future together with the other WG leads.

**Review by elft3r — APPROVED** [review]
LGTM

**Review by roberthstrand — APPROVED** [review]

**Pull request merged**